scripts/analyze_benchmarks/markdown.ts

```diff
@@ -0,0 +1,217 @@
+</tr>`;
+}
+
+function generateStatsTableRows(
+    results: StableAndExperimentalStatistics,
+    comparisonResults: StableAndExperimentalStatistics
+): string {
+    return `<tr>
+    <th></th>
+    <th>Number of Instructions</th>
+    <th></th>
+    <th>Number of Instructions</th>
+    <th></th>
+</tr>
+${Object.entries(results.stable)
+    .filter(
+        ([key]) => key !== 'count' && key !== 'baselineWeightedEfficiencyScore'
+    )
+    .map(([key, value]) => {
+        const statsKey = key as keyof Statistics;
+        const stableChange = comparisonResults.stable[statsKey];
+        const experimentalChange = comparisonResults.experimental[statsKey];
+        const metric = camelToTitleCase(key);
+        const stableFormattedValue = formatNumber(Math.floor(value));
+        const experimentalFormattedValue = formatNumber(
+            Math.floor(results.experimental[statsKey])
+        );
+        const stableFormattedChange = formatChangeValue(stableChange);
+        const experimentalFormattedChange =
+            formatChangeValue(experimentalChange);
+
+        return `<tr>
+    <td>${metric}</td>
+    <td>${stableFormattedValue}</td>
+    <td>${stableFormattedChange}</td>
+    <td>${experimentalFormattedValue}</td>
+    <td>${experimentalFormattedChange}</td>
+</tr>`;
+    })
+    .join('\n')}
+`;
+}
+
+function formatChangeValue(change: number): string {
+    if (change === 0) {
+        return `${change.toFixed(2)}%`;
+    }
+
+    return `<span style="color: ${change < 0 ? 'green' : 'red'}">${change.toFixed(2)}%</span>`;
+}
+
+function compareChanges(
+    results: Record<string, StableAndExperimentalStatistics>,
+    index: number
+): StableAndExperimentalStatistics {
+    const versions = Object.keys(results);
+    if (index + 1 < versions.length) {
+        const previous = versions[index + 1];
+        const current = versions[index];
+        const stable = calculateVersionChanges(
+            results[previous].stable,
+            results[current].stable
+        );
+        const experimental = calculateVersionChanges(
+            results[previous].experimental,
+            results[current].experimental
+        );
+        return { stable, experimental };
+    }
+
+    return {
+        stable: createDefaultStatistics(),
+        experimental: createDefaultStatistics()
+    };
+}
+
+function createDefaultStatistics(): Statistics {
+    return {
+        baselineWeightedEfficiencyScore: 0,
+        mean: 0,
+        median: 0,
+        min: 0,
+        count: 0,
+        standardDeviation: 0,
+        max: 0
+    };
+}
+
+function calculateChange(prevValue: number, currValue: number): number {
+    return ((currValue - prevValue) / prevValue) * 100;
+}
+
+function calculateVersionChanges(
+    previous: Statistics,
+    current: Statistics
+): Statistics {
+    return Object.keys(previous).reduce((changes, key) => {
+        const typedKey = key as keyof Statistics;
```

Comment:
Are all of these casts necessary?

Reply:
yes or else it will think it's just any old string